src/db/db-helper.js

```diff
@@ -241,22 +245,92 @@ module.exports = class DBHelper {
 
   static async updateEventStatusWithdrawing(currBlockTime) {
     try {
-      await db.Events.update(
+      const updatedEvents = await db.Events.update(
         {
           txStatus: TX_STATUS.SUCCESS,
           $not: { status: EVENT_STATUS.WITHDRAWING },
           arbitrationEndTime: { $lte: currBlockTime },
           currentRound: { $gt: 0 },
         },
         { $set: { status: EVENT_STATUS.WITHDRAWING } },
-        { multi: true },
+        {
+          multi: true,
+          returnUpdatedDocs: true,
+        },
       );
+      await DBHelper.updateLeaderboardWithdrawing(updatedEvents);
     } catch (err) {
       logger.error(`UPDATE Event Status Withdrawing error: ${err.message}`);
       throw err;
     }
   }
 
+  static async updateLeaderboardWithdrawing(events) {
+    try {
+      each(events, async (event) => {
+        // get all the participants
+        const bets = await DBHelper.findBet({ eventAddress: event.address });
+        const filtered = [];
+        each(bets, (bet) => {
+          if (!find(filtered, {
+            eventAddress: bet.eventAddress,
+            betterAddress: bet.betterAddress,
+          })) {
+            filtered.push(bet);
+          }
+        });
+        // calculate winnings for all participants
+        const promises = [];
+        const winnings = [];
+        each(filtered, (bet) => {
+          promises.push(new Promise(async (resolve, reject) => {
```

Comment:
this needs to be batched up. our RPC server maxes out at ~40 requests per second right now. if you were parsing 4 events x 10 winners per event, then you'd max out.

I am using `pLimit` (https://github.com/sindresorhus/p-limit) to handle this case. take a look how its currently implemented right now. use the same `limit` var in `sync/index`. this will work out when you move this logic into its own sync file.

https://github.com/bodhiproject/bodhi-server/blob/dev-062419/src/sync/index.js#L31
https://github.com/bodhiproject/bodhi-server/blob/dev-062419/src/sync/index.js#L183
https://github.com/bodhiproject/bodhi-server/blob/dev-062419/src/sync/bet-placed.js#L24